### PR TITLE
EWPP-583: Render iframe media type inside media container

### PR DESCRIPTION
**Pull request by yenyasinn**

### CODE REVIEW

File: modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaFormatter.php [hunk 1]
```diff
@@ -175,41 +176,7 @@ protected function viewElement(FieldItemInterface $item, string $langcode): arra
...
-      if ($media->bundle() === 'video_iframe') {
-        $ratio = $media->get('oe_media_iframe_ratio')->value;
-        $params['ratio'] = str_replace('_', '-', $ratio);
-      }
-
-      $build['#params'] = $params;
-      $cacheability->applyTo($build);
-
-      return $build;
+      return $this->viewEmbededMedia($media, $build, $params, $cacheability);
```
Comment (yenyasinn) [inline modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaFormatter.php:179]: It was rewritten due to cyclomatic complexity error in PhpMd

**Review by georgebaev — APPROVED** [review]

File: tests/Kernel/MediaRenderTest.php [hunk 2]
```diff
@@ -250,4 +254,31 @@ public function testDocumentMedia(): void {
...
+      'bundle' => 'iframe',
+      'name' => 'test iframe',
+      'oe_media_iframe' => '<iframe src="http://example.com/iframe_media"></iframe>',
+    ]);
+    $media->save();
+
+    $build = $this->mediaViewBuilder->view($media, 'oe_theme_main_content');
+    $html = $this->renderRoot($build);
+    $crawler = new Crawler($html);
+    $iframe = $crawler->filter('.ecl-media-container .ecl-media-container__media--ratio-16-9 iframe');
```
Comment (brummbar) [inline tests/Kernel/MediaRenderTest.php:272]: As per specifications, if no ratio is specified in the iframe media, no ratio should be used in the media container.

**Review by brummbar — CHANGES_REQUESTED** [review]
Early quick review, I didn't go through all the code.

File: tests/Kernel/MediaRenderTest.php (hunk 2 above)
Comment (yenyasinn) [reply tests/Kernel/MediaRenderTest.php:272]: ECL component components/ec-component-media-container/ecl-media-container.html.twig sets default ratio 16:9 if ratio isn't provided.
There isn't class that defines media container without ratio.
If we pass in ratio variable value `no-ratio` then class is `ecl-media-container__media--ratio-no-ratio` and media container looks broken:
Screenshot 2020-12-10 at 11 12 43

Therefore if we want not to set any ratio ECL's media container should be adjusted.

Comment (brummbar) [reply tests/Kernel/MediaRenderTest.php:272]: Well if the media container is meant only to handle responsiveness, then it's as easy as not using it when a custom iframe size is needed. Let's change the code this way.

Comment (yenyasinn) [reply tests/Kernel/MediaRenderTest.php:272]: @brummbar sorry, but what does "Let's change the code this way." mean?

**Comment by yenyasinn** [comment]
Closed in favour https://github.com/openeuropa/oe_theme/pull/690

**Pull request closed without merging**